lib/tcpip/FreeRTOS_errno_TCP.h

```diff
@@ -0,0 +1,54 @@
+// Copyright SCI Semiconductor and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+/**
+ * This header defines errno values used by FreeRTOS+TCP.
+ *
+ * `FreeRTOS_errno_TCP.h` used to be part of the FreeRTOS+TCP tree, but was
+ * removed because integrated in the FreeRTOS core, which we do not use. We
+ * therefore re-introduce it here.
```

Comment:
We should probably add this in the FreeRTOS compat layer.  It would also be good to define these in terms of the ones in `errno.h`, rather than as constant literals.